### PR TITLE
Create POA transfer agreement

### PR DESCRIPTION
**Pull request by TopSwagCode**

### CODE REVIEW

File: domains/transfer/Transfer.API/API.IntegrationTests/Transfer/Api/Controllers/TransferAgreementsControllerTest.cs [hunk 1]
```diff
@@ -580,6 +580,63 @@ await SeedTransferAgreements(
...
     }
 
+    [Fact]
+    public async Task CreatePOATransferAgreement_ShouldCreateTransferAgreement_WhenInputIsValid()
+    {
+        var sub = Guid.NewGuid();
+        var orgId = Any.OrganizationId();
+        var senderName = "Some Awesome Org";
+
+        var request = new CreateTransferAgreementRequest(sub, orgId.Value, DateTimeOffset.UtcNow.ToUnixTimeSeconds(), DateTimeOffset.UtcNow.ToUnixTimeSeconds(), "12345678", "lol", "87456123", senderName);
```
Comment (martinhenningjensen) [inline domains/transfer/Transfer.API/API.IntegrationTests/Transfer/Api/Controllers/TransferAgreementsControllerTest.cs:590]: ReceiverOrganizationId = sub and Guid SenderOrganizationId = own org id???

File: domains/transfer/Transfer.API/API.IntegrationTests/Transfer/Api/Controllers/TransferAgreementsControllerTest.cs [hunk 2]
```diff
@@ -580,6 +580,63 @@ await SeedTransferAgreements(
...
+        using var scope = factory.Services.CreateScope();
+        using var dbContext = scope.ServiceProvider.GetService<ApplicationDbContext>()!;
+
+        var authenticatedClient = factory.CreateB2CAuthenticatedClient(sub, orgId.Value);
+        var response = await authenticatedClient.PostAsJsonAsync($"api/transfer/transfer-agreements/create", request);
+
+        var transferAgreement = dbContext.TransferAgreements.Single(x => x.SenderId == orgId);
+
+        response.StatusCode.Should().Be(HttpStatusCode.Created);
+        transferAgreement.SenderName.Value.Should().Be(senderName);
```
Comment (martinhenningjensen) [inline domains/transfer/Transfer.API/API.IntegrationTests/Transfer/Api/Controllers/TransferAgreementsControllerTest.cs:600]: Assert on sender and receiver? (Are they the most important properties?)

File: domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementsController.cs [hunk 3]
```diff
@@ -270,4 +272,68 @@ private TransferAgreementStatus GetTransferAgreementStatusFromAgreement(Transfer
 
         return TransferAgreementStatus.Inactive;
     }
+
+    [HttpPost("create/")]
+    [ProducesResponseType(typeof(TransferAgreementDto), 200)]
+    [ProducesResponseType(typeof(void), 404)]
+    public async Task<ActionResult> CreateTransferAgreementDirectly([FromServices] IProjectOriginWalletClient walletClient, [FromBody] CreateTransferAgreementRequest request)
+    {
+        accessDescriptor.IsAuthorizedToOrganizations([request.SenderOrganizationId, request.ReceiverOrganizationId]);
```
Comment (martinhenningjensen) [inline domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementsController.cs:281]: Why no command?

File: domains/transfer/Transfer.API/API/Transfer/Api/Repository/ITransferAgreementRepository.cs [hunk 4]
```diff
@@ -10,6 +10,7 @@ public interface ITransferAgreementRepository
 {
     Task<TransferAgreement> AddTransferAgreementToDb(TransferAgreement transferAgreement, CancellationToken cancellationToken);
     Task<TransferAgreement> AddTransferAgreementAndDeleteProposal(TransferAgreement newTransferAgreement, Guid proposalId, CancellationToken cancellationToken);
+    Task<TransferAgreement> AddTransferAgreement(TransferAgreement newTransferAgreement, CancellationToken cancellationToken);
```
Comment (martinhenningjensen) [inline domains/transfer/Transfer.API/API/Transfer/Api/Repository/ITransferAgreementRepository.cs:13]: Isn't this the same as method _AddTransferAgreementToDb_? (Line 11 above)

**Review by martinhenningjensen — CHANGES_REQUESTED** [review]

File: domains/transfer/Transfer.API/API.IntegrationTests/Transfer/Api/Controllers/TransferAgreementsControllerTest.cs (hunk 1 above)
Comment (TopSwagCode) [reply domains/transfer/Transfer.API/API.IntegrationTests/Transfer/Api/Controllers/TransferAgreementsControllerTest.cs:590]: yes. Will rename to simplify reading of test :)

File: domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementsController.cs (hunk 3 above)
Comment (TopSwagCode) [reply domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementsController.cs:281]: I have gone commando

**Pull request merged**